### PR TITLE
 Stack: Making 'gap' prop be able to take two values, one for horizontal gap and one for vertical gap, and removing 'verticalGap' prop

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

- Making `gap` prop in `IStackProps` be able to take two values, one for horizontal gap and one for vertical gap, in an effort to follow the [gap CSS property specification](https://developer.mozilla.org/en-US/docs/Web/CSS/gap):

> This property is specified as a value for `<'row-gap'>` followed optionally by a value for `<'column-gap'>`. If `<'column-gap'>` is omitted, it’s set to the same value as `<'row-gap'>`.

- Removing `verticalGap` prop as result of previous changes.

- Updating tests and examples that used `gap` and `verticalGap` props.

- Adding new tests for `StackUtils`.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7828)

### CODE REVIEW

File: packages/experiments/src/components/Stack/Stack.types.ts [hunk 1]
```diff
@@ -80,15 +80,12 @@ export interface IStackProps
   grow?: boolean | number | 'inherit' | 'initial' | 'unset';
 
   /**
-   * Defines the horizontal spacing between Stack child elements.
+   * Defines the spacing between Stack children elements.
+   * The property is specified as a value for 'horizontal gap', followed optionally by a value for 'vertical gap'.
+   * If 'vertical gap' is omitted, it's set to the same value as 'horizontal gap'.
    */
   gap?: number | string;
```
Comment (micahgodbolt) [inline packages/experiments/src/components/Stack/Stack.types.ts]: grid gap is "row gap then column gap"
image


so if we are calling it vertical and horizontal then it should be "vertical horizontal"

Comment (micahgodbolt) [reply packages/experiments/src/components/Stack/Stack.types.ts]: actually i'm trying to figure out if you have this right or not. i think the issue is vertical/horizontal.....not sure if horizontal gap is gap between horizontal items, or gap running horizontally

can we switch to row and column gap?

Comment (khmakoto) [reply packages/experiments/src/components/Stack/Stack.types.ts]: That's right. Hmmmm, do we want to have it as 'vertical horizontal' though? Doesn't matter that much to me, but given that we had the original `gap` as the horizontal one I want to raise the question.

CC'ing @dzearing to get his opinion.

Comment (micahgodbolt) [reply packages/experiments/src/components/Stack/Stack.types.ts]: I 'think' the order is right, but naming convention is confusing. What is 'vertical gap'? distance between items stacked vertically...or the gap that runs vertically? I'm mostly wanting to make sure we are using the same approach as grid is, and suggesting we use their terminology when talking about the gap. Column gap vs horizontal/vertical.

Comment (khmakoto) [reply packages/experiments/src/components/Stack/Stack.types.ts]: Fixed in latest commit 👍

File: packages/experiments/src/components/Stack/Stack.styles.ts [hunk 2]
```diff
@@ -32,14 +31,7 @@ export const styles: IStackComponent['styles'] = (props, theme): IStackStylesRet
...
 
-  let horiGap: IStackProps['gap'];
-  let vertGap: IStackProps['gap'];
-
-  horiGap = gap;
-  vertGap = verticalGap !== undefined ? verticalGap : gap;
-
-  const hGap = parseGap(horiGap, theme);
-  const vGap = parseGap(vertGap, theme);
+  const { vGap, hGap } = parseGap(gap, theme);
```
Comment (micahgodbolt) [inline packages/experiments/src/components/Stack/Stack.styles.ts]: should we use rGap and cGap internally? or even rowGap columnGap for clarify?

**Comment by micahgodbolt** [comment]
looks like we need to update the screener tests to use the new gap (and remove verticalGap)

**Comment by micahgodbolt** [comment]
@khmakoto loving the improved wording in the type description :shipit:

**Pull request merged**